File: deal-machine-mcp/test-api.js
// Quick API test script
import axios from 'axios';

const API_KEY = process.env.DEALMACHINE_API_KEY || 'your_api_key_here';
const BASE_URL = 'https://api.dealmachine.com/public/v1';

async function testEndpoints() {
  const client = axios.create({
    baseURL: BASE_URL,
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json',
      'Accept': 'application/json',
    },
    timeout: 10000,
    maxRedirects: 5,
    validateStatus: (status) => status < 500, // Don't throw on 4xx errors
  });

  console.log('Testing Deal Machine API endpoints...\n');

  // Test 1: Team members (known to work)
  try {
    console.log('1. Testing GET /team-members/');
    const response = await client.get('/team-members/');
    console.log('✅ SUCCESS:', response.status);
    console.log('Response:', JSON.stringify(response.data, null, 2).substring(0, 500));
    console.log('\n');
  } catch (error) {
    console.log('❌ FAILED:', error.response?.status, error.response?.data || error.message);
    console.log('\n');
  }

  // Test 2: Get leads
  try {
    console.log('2. Testing GET /leads/');
    const response = await client.get('/leads/');
    console.log('✅ SUCCESS:', response.status);
    console.log('Response:', JSON.stringify(response.data, null, 2).substring(0, 500));
    console.log('\n');
  } catch (error) {
    console.log('❌ FAILED:', error.response?.status, error.response?.data || error.message);
    console.log('\n');
  }

  // Test 3: Get lead statuses
  try {
    console.log('3. Testing GET /lead-statuses/');
    const response = await client.get('/lead-statuses/');
    console.log('✅ SUCCESS:', response.status);
    console.log('Response:', JSON.stringify(response.data, null, 2).substring(0, 500));
    console.log('\n');
  } catch (error) {
    console.log('❌ FAILED:', error.response?.status, error.response?.data || error.message);
    console.log('\n');
  }

  // Test 4: Get custom fields
  try {
    console.log('4. Testing GET /custom-fields/');
    const response = await client.get('/custom-fields/');
    console.log('✅ SUCCESS:', response.status);
    console.log('Response:', JSON.stringify(response.data, null, 2).substring(0, 500));
    console.log('\n');
  } catch (error) {
    console.log('❌ FAILED:', error.response?.status, error.response?.data || error.message);
    console.log('\n');
  }

  // Test 5: Get tags
  try {
    console.log('5. Testing GET /tags/');
    const response = await client.get('/tags/');
    console.log('✅ SUCCESS:', response.status);
    console.log('Response:', JSON.stringify(response.data, null, 2).substring(0, 500));
    console.log('\n');
  } catch (error) {
    console.log('❌ FAILED:', error.response?.status, error.response?.data || error.message);
    console.log('\n');
  }

  console.log('API testing complete!');
}

testEndpoints().catch(console.error);

File: deal-machine-agent/test_api.py
#!/usr/bin/env python3
"""
DealMachine API Test Script
Tests connection and basic functionality with the DealMachine API
"""

import os
import sys
import json
import requests
from datetime import datetime

# Load environment variables
API_KEY = os.getenv('DEALMACHINE_API_KEY', 'nqcymtWdwB16')
BASE_URL = os.getenv('DEALMACHINE_API_BASE_URL', 'https://api.dealmachine.com/public/v1')

def test_api_connection():
    """Test basic API connectivity"""
    print("=" * 60)
    print("DEALMACHINE API CONNECTION TEST")
    print("=" * 60)
    print(f"API Key: {API_KEY[:10]}..." if len(API_KEY) > 10 else API_KEY)
    print(f"Base URL: {BASE_URL}")
    print(f"Time: {datetime.now()}")
    print("=" * 60)

    headers = {
        'Authorization': f'Bearer {API_KEY}',
        'Content-Type': 'application/json'
    }

    # Test 1: Get Team Members (simplest endpoint)
    print("\n[TEST 1] Fetching Team Members...")
    try:
        response = requests.get(
            f'{BASE_URL}/team-members/',
            headers=headers,
            timeout=10
        )
        print(f"Status Code: {response.status_code}")

        if response.status_code == 200:
            data = response.json()
            print("✅ SUCCESS - Team Members Endpoint")
            print(f"Response: {json.dumps(data, indent=2)[:500]}...")
        elif response.status_code == 401:
            print("❌ FAILED - Invalid API Key (401 Unauthorized)")
            print(f"Response: {response.text}")
            return False
        else:
            print(f"⚠️  WARNING - Unexpected status code: {response.status_code}")
            print(f"Response: {response.text}")
    except Exception as e:
        print(f"❌ ERROR: {str(e)}")
        return False

    # Test 2: Get Leads
    print("\n[TEST 2] Fetching Leads...")
    try:
        response = requests.get(
            f'{BASE_URL}/leads/?page=1&limit=5',
            headers=headers,
            timeout=10
        )
        print(f"Status Code: {response.status_code}")

        if response.status_code == 200:
            data = response.json()
            print("✅ SUCCESS - Leads Endpoint")

            if isinstance(data, dict) and 'leads' in data:
                leads = data.get('leads', [])
                print(f"Found {len(leads)} leads")
                if leads:
                    print(f"First lead: {json.dumps(leads[0], indent=2)[:300]}...")
            else:
                print(f"Response: {json.dumps(data, indent=2)[:500]}...")
        else:
            print(f"⚠️  Status {response.status_code}: {response.text[:200]}")
    except Exception as e:
        print(f"❌ ERROR: {str(e)}")

    # Test 3: Get Tags
    print("\n[TEST 3] Fetching Tags...")
    try:
        response = requests.get(
            f'{BASE_URL}/tags/',
            headers=headers,
            timeout=10
        )
        print(f"Status Code: {response.status_code}")

        if response.status_code == 200:
            data = response.json()
            print("✅ SUCCESS - Tags Endpoint")

            if isinstance(data, dict) and 'tags' in data:
                tags = data.get('tags', [])
                print(f"Found {len(tags)} tags")
                if tags:
                    print(f"Sample tags: {json.dumps(tags[:3], indent=2)}")
            else:
                print(f"Response: {json.dumps(data, indent=2)[:500]}...")
        else:
            print(f"⚠️  Status {response.status_code}: {response.text[:200]}")
    except Exception as e:
        print(f"❌ ERROR: {str(e)}")

    print("\n" + "=" * 60)
    print("TEST COMPLETE")
    print("=" * 60)

    return True

if __name__ == '__main__':
    success = test_api_connection()
    sys.exit(0 if success else 1)
